learning/step_by_step/godot_design_philosophy.rst

```diff
@@ -0,0 +1,145 @@
+
+|image2|
+
+The goal is to offer a full package to create games and a continuous
+user experience. You can still work with external programs as long as
+there is an import plugin for it or you can create one, like the `Tiled
+map editor importer <https://github.com/vnen/godot-tiled-importer>`__.
+
+That is also partly why Godot offers its own programming languages
+GDscript and VisualScript, along with C#. They’re designed for the needs
+of game developers, game designers, and they’re tightly integrated in
+the engine and the editor.
+
+|image3|
+
+*VisualScript is a node-based programming language that integrates well
+in the editor. You can drag and drop nodes or resources into the graph
+to create new code blocks.*
+
+Note that if this is true for 2d at the time of writing, the 3d
+workspace still doesn’t feature as many tools. You’ll need external
+programs or add-ons to edit terrains, animate complex characters, etc.
+Godot provides a complete API to extend the editor’s functionality using
+game code. See `The Godot editor is a Godot game <#>`__ below.
+
+|image4|
+
+*A State Machine editor plugin in Godot 2 by kubecz3k. It lets you
+manage states and transitions visually*
+
+Open source
+-----------
+
+Godot offers a fully open source code-base under the MIT license. This
+means all the technologies that ship with it have to be Free as well.
+For the most part, they’re coded from the ground up by contributors.
+
+Anyone can plug in proprietary tools for the needs of their projects.
+They just won’t ship with the engine. This may include NViDia PhysX,
+Google Admob or an FBX file importer. Any of these can come as
+third-party plugins instead.
+
+On the other hand, an open codebase means you can learn from and extend
+the engine to your heart’s content.
+
+Note this does not affect the work you do with Godot in any way: there’s
+no strings attached to the engine or anything you make with it.
+
+Community-driven
+----------------
+
+**Godot is made by its community, for the community and all game
+creators out there**. It’s the needs of the users and open discussions
+that drive the core updates. New features from the core developers often
+focus on what will benefit the most users first.
+
+That said, although a handful of core developers work on it full-time,
+the project has over 500 contributors at the time of writing. Benevolent
+programmers work on features they may need themselves so you’ll see
+improvements in all corners of the engine at the same time in every
+major release.
+
+The Godot editor is a Godot game
+--------------------------------
+
+The Godot editor runs on the game engine. It uses the engine’s own UI
+system, it can hot reload code and scenes when you test your projects,
+or run game code in the editor. This means you can **use the same code**
+and scenes for your games or **to build plugins and extend the editor**.
+
+This leads to a reliable and flexible UI system as it powers the editor
+itself.
+
```

Comment:
Drop a mention of `Tool` scripts and `EditorPlugin`s and how easy it is to extend the editor for your workflow, was hinted at above/previously but could be more IMO